semgrep/semgrep/rule_schema.yaml

```diff
@@ -127,6 +135,7 @@ definitions:
     type: object
     properties:
       pattern-not:
+        title: (deprecated, use pattern-not-inside) Do not return finding where Semgrep pattern matches exactly
```

Comment:
Is this deprecated? I don't think we have that language anywhere else

Reply:
Discussed many times before, thought I'd just document it as such and let a discussion start here :D

Reply:
cc @dlukeomalley

Reply:
I suspect that discussion will greatly slow down these unrelated changes. We'll probably need to figure out a deprecation plan, docs updates, etc, if we do in fact want to deprecate `pattern-not`. If you want to move quickly with these changes it'd probably be easiest to keep `pattern-not` deprecation separate :man_shrugging: